php-checks/src/main/java/org/sonar/php/checks/DuplicatedMethodCheck.java

```diff
@@ -0,0 +1,123 @@
+
+@Rule(key = "S4144")
+public class DuplicatedMethodCheck extends PHPVisitorCheck {
+
+  private static final String ISSUE_MSG = "Update this method so that its implementation is not identical to \"%s\" on line %d.";
+  private static final Function<FunctionTree, NameIdentifierTree> METHOD_TO_NAME = f -> ((MethodDeclarationTree) f).name();
+  private static final Function<FunctionTree, NameIdentifierTree> FUNCTION_TO_NAME = f -> ((FunctionDeclarationTree) f).name();
+  private final Deque<List<MethodDeclarationTree>> methods = new LinkedList<>();
+  private List<FunctionDeclarationTree> functions = new ArrayList<>();
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    super.visitCompilationUnit(tree);
+    checkDuplications(functions, FUNCTION_TO_NAME);
+  }
+
+  @Override
+  public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
+    // Ignore empty functions
+    if (!tree.body().statements().isEmpty()) {
+      functions.add(tree);
+    }
+    super.visitFunctionDeclaration(tree);
+  }
+
+  @Override
+  public void visitMethodDeclaration(MethodDeclarationTree tree) {
+    // Ignore abstract and empty methods
+    if (isDuplicateCandidate(tree)) {
+      methods.peek().add(tree);
+    }
+    super.visitMethodDeclaration(tree);
+  }
+
+  private static boolean isDuplicateCandidate(MethodDeclarationTree tree) {
+    return tree.body().is(Tree.Kind.BLOCK) && (((BlockTree) tree.body()).statements().size() >=2 || isAccessor(tree));
+  }
+
+  private static boolean isAccessor(MethodDeclarationTree tree) {
+    return ((BlockTree) tree.body()).statements().size() == 1
+      && (tree.name().text().startsWith("set") || tree.name().text().startsWith("get") || tree.name().text().startsWith("is"));
+  }
+
+  @Override
+  public void visitClassDeclaration(ClassDeclarationTree tree) {
+    methods.push(new ArrayList<>());
+    super.visitClassDeclaration(tree);
+    checkDuplications(methods.pop(), METHOD_TO_NAME);
+  }
+
+  @Override
+  public void visitAnonymousClass(AnonymousClassTree tree) {
+    methods.push(new ArrayList<>());
+    super.visitAnonymousClass(tree);
+    checkDuplications(methods.pop(), METHOD_TO_NAME);
+  }
+
+  private void checkDuplications(List<? extends FunctionTree> functionDeclarations, Function<FunctionTree, NameIdentifierTree> toName) {
+    Set<? super FunctionTree> reported = new HashSet<>();
+    for (int i = 0; i < functionDeclarations.size(); i++) {
+      FunctionTree func = functionDeclarations.get(i);
+      SyntaxToken methodIdentifier = toName.apply(func).token();
+      List<StatementTree> methodBody = ((BlockTree) func.body()).statements();
+      functionDeclarations.stream()
+        .skip(i + 1L)
+        // avoid reporting multiple times
+        .filter(m -> !reported.contains(m))
+        // skip overloads
+        .filter(m -> !methodIdentifier.text().equals(toName.apply(m).text()))
```

Comment:
We should add a test to check that.

Reply:
in fact overload are not possible in PHP, so the line has been dropped alltogether.